phpBB/phpbb/template/twig/node/event.php

```diff
@@ -42,40 +46,80 @@ public function compile(\Twig\Compiler $compiler)
 
 		$location = $this->listener_directory . $this->getNode('expr')->getAttribute('name');
 
+		$compiler_steps = [];
+
 		foreach ($this->environment->get_phpbb_extensions() as $ext_namespace => $ext_path)
 		{
 			$ext_namespace = str_replace('/', '_', $ext_namespace);
 
+			$compiler_calls = [];
+
 			if ($this->environment->isDebug())
 			{
 				// If debug mode is enabled, lets check for new/removed EVENT
 				//  templates on page load rather than at compile. This is
 				//  slower, but makes developing extensions easier (no need to
 				//  purge the cache when a new event template file is added)
-				$compiler
-					->write("if (\$this->env->getLoader()->exists('@{$ext_namespace}/{$location}.html')) {\n")
-					->indent()
-				;
+				$compiler_calls[] = function() use($compiler, $ext_namespace, $location) {
+					$compiler
+						->write("if (\$this->env->getLoader()->exists('@{$ext_namespace}/{$location}.html')) {\n")
+						->indent()
+					;
+				};
 			}
 
 			if ($this->environment->isDebug() || $this->environment->getLoader()->exists('@' . $ext_namespace . '/' . $location . '.html'))
 			{
-				$compiler
-					->write("\$previous_look_up_order = \$this->env->getNamespaceLookUpOrder();\n")
-
-					// We set the namespace lookup order to be this extension first, then the main path
-					->write("\$this->env->setNamespaceLookUpOrder(array('{$ext_namespace}', '__main__'));\n")
-					->write("\$this->env->loadTemplate('@{$ext_namespace}/{$location}.html')->display(\$context);\n")
-					->write("\$this->env->setNamespaceLookUpOrder(\$previous_look_up_order);\n")
-				;
+				$compiler_calls[] = function() use($compiler, $ext_namespace, $location) {
+					$compiler
+						->write("\$previous_look_up_order = \$this->env->getNamespaceLookUpOrder();\n")
+
+						// We set the namespace lookup order to be this extension first, then the main path
+						->write("\$this->env->setNamespaceLookUpOrder(array('{$ext_namespace}', '__main__'));\n")
+						->write("\$this->env->loadTemplate('@{$ext_namespace}/{$location}.html')->display(\$context);\n")
+						->write("\$this->env->setNamespaceLookUpOrder(\$previous_look_up_order);\n")
+					;
+				};
 			}
 
 			if ($this->environment->isDebug())
 			{
-				$compiler
-					->outdent()
-					->write("}\n\n")
-				;
+				$compiler_calls[] = function() use($compiler) {
```

Comment:
It seems that this call is intended to be the last in the queue but will be messed up later with `krsort()`; so that won't kept the last call.